### PR TITLE
feat: Added resource browser page, listing & details views

### PR DESCRIPTION
**Pull request by sohel-rp**
# Description

This contain changes for,
1. Navigation to the resource browser page
2. Cluster list page
3. Left side resource list
4. Resource details panel
5. Updates in the resource tab component & implementation on resource browser page
6. Updates in node details component & implementation on resource browser page
- Manifest
- Events
- Logs
- Terminal
7. And some related changes

Fixes - [AB#1616](https://dev.azure.com/DevtronLabs/Devtron/_workitems/edit/1616) & [AB#1652](https://dev.azure.com/DevtronLabs/Devtron/_workitems/edit/1652)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

All tests have been performed manually by visiting the new resource browser section & performing different options. Also, sanity has also been done on the app details page as some of the components are re-used from it in the resource browser.

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/v2/appDetails/k8Resource/NodeTreeTabList.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+                                            tab.isSelected ? 'resource-tree-tab bcn-0 cn-9' : ''
+                                        } flex left pl-12 pt-8 pb-8 pr-12`}
+                                    >
+                                        {getTabNavLink(tab)}
+                                        {tab.name !== AppDetailsTabs.log_analyzer &&
+                                            tab.name !== AppDetailsTabs.k8s_Resources && (
+                                                <div className="resource-tab__close-wrapper flex br-5">
+                                                    <Cross
+                                                        onClick={(e) => handleCloseTab(e, tab.title)}
+                                                        className="icon-dim-16 cursor"
```
Comment (vivek-devtron) [inline src/components/v2/appDetails/k8Resource/NodeTreeTabList.tsx:119]: we can extract this inline method

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx [hunk 2]
```diff
@@ -1,138 +1,229 @@
...
+    const params = useParams<{ actionName: string; podName: string; nodeType: string; node: string }>()
+    const [tabs, setTabs] = useState([])
+    const [selectedTabName, setSelectedTabName] = useState('')
+    const [resourceContainers, setResourceContainers] = useState([])
+    const [isResourceDeleted, setResourceDeleted] = useState(false)
+    const [fetchingResource, setFetchingResource] = useState(false)
+    const { path, url } = useRouteMatch()
+
+    useEffect(() => {
+        const _tabs = getNodeDetailTabs(params.nodeType as NodeType)
```
Comment (vivek-devtron) [inline src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx:41]: we should add a check for the dependency

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx [hunk 3]
```diff
@@ -1,138 +1,229 @@
...
+                }
+
+                if (Array.isArray(result.manifest.spec.initContainers)) {
+                    _resourceContainers.push(...result.manifest.spec.containers.map((_container) => _container.name))
+                }
+            }
+            setResourceContainers(_resourceContainers)
+
+            // Clear out error on node change
+            if (isResourceDeleted) {
```
Comment (vivek-devtron) [inline src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx:75]: should we do this in error case as well?

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx [hunk 4]
```diff
@@ -1,138 +1,229 @@
...
-                <Route
-                    path={`${path}/${NodeDetailTab.TERMINAL}`}
-                    render={() => {
-                        return <TerminalComponent selectedTab={handleSelectedTab} isDeleted={isDeleted} />;
-                    }}
+                    })}
+            </div>
+            {fetchingResource || (isResourceBrowserView && (loadingResources || !selectedResource)) ? (
+                <MessageUI
+                    msg={''}
```
Comment (vivek-devtron) [inline src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx:169]: we can directly pass ""

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/Events.component.tsx [hunk 5]
```diff
@@ -1,64 +1,76 @@
...
+            setLoading(false)
         }
-    }, [params.podName, params.nodeType]);
+    }, [params.podName, params.node, params.nodeType])
 
     return (
-        <div style={{ minHeight: '600px', background: '#0B0F22', flex: 1 }}>
+        <div
+            className="events-table-container"
+            style={{ minHeight: isResourceBrowserView ? '200px' : '600px', background: '#0B0F22', flex: 1 }}
```
Comment (vivek-devtron) [inline src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/Events.component.tsx:56]: use color name of `#0B0F22`

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/EventsTable.tsx [hunk 6]
```diff
@@ -4,16 +4,25 @@ import { MESSAGING_UI } from '../../../../../../config/constants'
...
+                />
+            )
         } else {
             if (eventsList && eventsList.length > 0) {
                 return (
                     <div className="cn-0 ">
                         <table className="table pl-20">
-                            <thead style={{ minHeight: '600px', background: '#0B0F22' }}>
+                            <thead
+                                style={{ minHeight: isResourceBrowserView ? '200px' : '600px', background: '#0B0F22' }}
```
Comment (vivek-devtron) [inline src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/EventsTable.tsx:24]: same here

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx (hunk 4 above)
Comment (sohel-rp) [reply src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx]: done

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx (hunk 3 above)
Comment (sohel-rp) [reply src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx:85]: done

File: src/components/v2/appDetails/k8Resource/NodeTreeTabList.tsx (hunk 1 above)
Comment (sohel-rp) [reply src/components/v2/appDetails/k8Resource/NodeTreeTabList.tsx:124]: done

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx (hunk 2 above)
Comment (sohel-rp) [reply src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx]: done

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/EventsTable.tsx (hunk 6 above)
Comment (sohel-rp) [reply src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/EventsTable.tsx]: done

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**